core/controllers/profile_test.py

```diff
@@ -791,6 +795,120 @@ def test_delete_delete_account_page_disabled(self):
+                    u'viewable_exploration_ids': [],
+                    u'owned_exploration_ids': [],
+                    u'voiced_exploration_ids': [],
+                    u'editable_exploration_ids': []
+                },
+                u'topic_snapshot_metadata_data': {},
+                u'completed_activities_data': {},
+                u'general_feedback_thread_data': {},
+                u'topic_rights_snapshot_metadata_data': {},
+                u'user_stats_data': {},
+                u'exploration_rights_snapshot_metadata_data': {},
+                u'user_subscriptions_data': {
+                    u'creator_ids': [],
+                    u'collection_ids': [],
+                    u'activity_ids': [],
+                    u'general_feedback_thread_ids': [],
+                    u'last_checked': None
+                },
+                u'config_property_snapshot_metadata_data': {},
+                u'exploration_snapshot_metadata_data': {},
+                u'incomplete_activities_data': {},
+                u'user_skill_mastery_data': {},
+                u'exp_user_last_playthrough_data': {},
+                u'user_settings_data': {
+                    u'username': u'editor',
+                    u'last_agreed_to_terms':
+                        python_utils.UNICODE(
+                            user_settings.last_agreed_to_terms),
+                    u'last_started_state_translation_tutorial': None,
+                    u'last_started_state_editor_tutorial': None,
+                    u'normalized_username': u'editor',
+                    u'first_contribution_msec': None,
+                    u'preferred_language_codes': [
+                        u'en'
+                    ],
+                    u'creator_dashboard_display_pref': u'card',
+                    u'subject_interests': [],
+                    u'default_dashboard': None,
+                    u'preferred_site_language_code': None,
+                    u'user_bio': u'',
+                    u'profile_picture_data_url':
+                        python_utils.UNICODE(
+                            user_settings.profile_picture_data_url),
+                    u'role': u'EXPLORATION_EDITOR',
+                    u'last_edited_an_exploration': None,
+                    u'email': u'editor@example.com',
+                    u'preferred_audio_language_code': None,
+                    u'last_logged_in':
+                        python_utils.UNICODE(
+                            user_settings.last_logged_in)
+                },
+                u'general_suggestion_data': {},
+                u'user_contribution_scoring_data': {},
+                u'general_feedback_email_reply_to_id_data': {},
+                u'collection_rights_snapshot_metadata_data': {}
+            }
+            self.assertEqual(
+                data,
+                expected_data
+            )
+
+    def test_export_account_handler_disabled(self):
```

Comment:
You should check three cases in three separate tests: request made when logged out and the functionality is enabled, request made when logged out and the functionality is disabled, and request made when logged in but the functionality is disabled.

Reply:
Done